sdk/core/Azure.Core/src/TokenCredential.cs

```diff
@@ -11,6 +12,27 @@ namespace Azure.Core
     /// </summary>
     public abstract class TokenCredential
     {
+        /// <summary>
+        /// Create the instance of a <see cref="TokenCredential"/>.
+        /// </summary>
+        protected TokenCredential() : this (TimeSpan.FromMinutes(2)) { }
+
+        /// <summary>
+        /// Create the instance of a <see cref="TokenCredential"/>.
+        /// </summary>
+        /// <param name="tokenRefreshOffset">A <see cref="TimeSpan"/> value representing the amount of time to subtract from the token expiry time, whereupon attempts will be made to refresh the token.</param>
+        protected TokenCredential(TimeSpan tokenRefreshOffset)
+        {
+            TokenRefreshOffset = tokenRefreshOffset;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="TimeSpan"/> value representing the amount of time to subtract from the token expiry time, whereupon
+        /// attempts will be made to refresh the token. By default this will occur two minutes prior to the expiry of the token.
+        /// </summary>
+        /// <returns>The duration value representing the amount of time to subtract from the token expiry time.</returns>
+        protected internal TimeSpan TokenRefreshOffset { get; }
```

Comment:
I think we had planned to make this return a structure or class `TokenRefreshOptions` rather than just a `TimeSpan` so it will be more forward compatible if we add more configuration around token refresh. This is in the java PR of this feature [here](https://github.com/Azure/azure-sdk-for-java/pull/12616/files#diff-1116664af60b3e486890e3b0312ba4aa)

Reply:
`TokenCredentialOptions` is defined in `Azure.Identity` while `TokenCredential` is defined in `Azure.Core`. I thought it would be a breaking change to move it.

Reply:
I wasn't saying the type would be `TokenCredentialOptions` but a new type `TokenRefreshOptions`

Reply:
So you want a struct that currently has one property `TokenRefreshOffset`, correct? Should we then add a public setter to the property?

Reply:
Fixed